src/main/java/org/sonatype/nexus/ci/util/RepositoryManagerClientUtil.groovy

```diff
@@ -27,14 +29,66 @@ import jenkins.model.Jenkins
 @SuppressWarnings(value = ['AbcMetric'])
 class RepositoryManagerClientUtil
 {
+  /**
+   * @deprecated use version specific call {@link RepositoryManagerClientUtil#nexus2Client}
+   */
+  @Deprecated
   static RepositoryManagerV2Client newRepositoryManagerClient(String url, String credentialsId)
       throws URISyntaxException
+  {
+    nexus2Client(url, credentialsId)
+  }
+
+  static RepositoryManagerV2Client nexus2Client(String url, String credentialsId)
+      throws URISyntaxException
+  {
+    def uri = new URI(url)
+    def clientBuilder = RepositoryManagerV2ClientBuilder.create().
+        withServerConfig(getServerConfig(uri, credentialsId))
+    def jenkinsProxy = Jenkins.instance.proxy
+
+    if (jenkinsProxy && ProxyUtil.shouldProxyForUri(jenkinsProxy, uri)) {
+      clientBuilder.withProxyConfig(ProxyUtil.newProxyConfig(jenkinsProxy))
+    }
+
+    return clientBuilder.build()
+  }
+
+  /**
+   * Creates a nxrm 3.x client that does not authenticate (anonymous access)
+   * @param url the nexus repository manager 3.x server url
+   * @return a {@link RepositoryManagerV3Client}
+   */
+  static RepositoryManagerV3Client nexus3Client(String url) {
+    nexus3Client(url, null, true)
+  }
+
+  /**
+   * Creates a nxrm 3.x client
+   * @param url the nexus repository manager 3.x server url
+   * @param credentialsId the id of the credentials configured in jenkisn
+   * @param anonymousAccess whether anonymous access is enabled on the nxrm 3.x server
+   * @return a {@link RepositoryManagerV3Client}
+   */
+  static RepositoryManagerV3Client nexus3Client(String url, String credentialsId, boolean anonymousAccessEnabled)
+      throws URISyntaxException
   {
     def uri = new URI(url)
+    def clientBuilder = RepositoryManagerV3ClientBuilder.create().
+        withServerConfig(getServerConfig(uri, credentialsId))
+    def jenkinsProxy = Jenkins.instance.proxy
 
+    if (jenkinsProxy && ProxyUtil.shouldProxyForUri(jenkinsProxy, uri)) {
+      clientBuilder.withProxyConfig(ProxyUtil.newProxyConfig(jenkinsProxy))
+    }
+
+    return clientBuilder.withAnonymousAccess(anonymousAccessEnabled).build()
+  }
+
+  private static ServerConfig getServerConfig(URI uri, String credentialsId) {
     def credentials = CredentialsMatchers.firstOrNull(CredentialsProvider
         .lookupCredentials(StandardUsernamePasswordCredentials, Jenkins.getInstance(), ACL.SYSTEM,
```

Comment:
I'm confused as to how it will become an issue if it isn't already; this particular line isn't new code, and if I understand correctly then this plugin is already in the wild with Jenkins folders. The only available NXRM action provided by the plugin does not offer an override for using different credentials than was configured for the target NXRM server in the global configuration which should be available from any folder?

Fully could be misunderstanding the situation, but will need additional context/info to be able to gauge if we can take on any additional work that would be required.

Reply:
This is most definitely an existing issue. If we want to promote (to our customers) staging via Jenkins, it's up to @mrprescott to determine whether or not you want to fix this before the promotion. See [0] where it was fixed for the IQ functionality and [1] for the existing bug.

[0] https://issues.sonatype.org/browse/INT-229
[1] https://issues.sonatype.org/browse/INT-416